File: js/packages/teams-ai/src/prompts/AssistantMessage.ts
/**
 * @module teams-ai
 */
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */

import { TemplateSection } from './TemplateSection';

/**
 * A message sent by the assistant.
 */
export class AssistantMessage extends TemplateSection {
    /**
     * Creates a new 'AssistantMessage' instance.
     * @param {string} template - Template to use for this section.
     * @param {number} tokens - Optional. Sizing strategy for this section. Defaults to -1, `auto`.
     * @param {string} assistantPrefix -  Optional. Prefix to use for assistant messages when rendering as text. Defaults to `assistant: `.
     */
    public constructor(template: string, tokens: number = -1, assistantPrefix: string = 'assistant: ') {
        super(template, 'assistant', tokens, true, '\n', assistantPrefix);
    }
}

File: python/packages/ai/teams/ai/prompts/assistant_message.py
"""
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the MIT License.
"""

from __future__ import annotations

from .sections.template_section import TemplateSection


class AssistantMessage(TemplateSection):
    """
    A message sent by the assistant.
    """

    def __init__(self, template: str, tokens: float = -1, text_prefix: str = "assistant: "):
        """
        Creates a new 'AssistantMessage' instance.

        Args:
            template (str): Template to use for this section.
            tokens (int, optional): Sizing strategy for this section. Defaults to `auto`.
            text_prefix (str, optional): Prefix to use for assistant messages when
              rendering as text. Defaults to `assistant: `.
        """
        super().__init__(template, "assistant", tokens, True, "\n", text_prefix)
